dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import com.google.common.reflect.ClassPath.ClassInfo;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.bcel.Const;
+import org.apache.bcel.classfile.Attribute;
+import org.apache.bcel.classfile.ConstantPool;
+import org.apache.bcel.classfile.InnerClass;
+import org.apache.bcel.classfile.InnerClasses;
+import org.apache.bcel.classfile.JavaClass;
+import org.apache.bcel.util.ClassPath;
+import org.apache.bcel.util.SyntheticRepository;
+
+/**
+ * This class reads jar files and runs static linkage check on them.
+ * Static linkage check finds discrepancies between methods referenced by classes in the jar files,
+ * and methods defined in them. This happens when the signature of a non-private method or class in
+ * a dependency has changed in an incompatible way between the version supplied at compile time
+ * and the version invoked at runtime.
+ * TODO: enhance scope to include fields and classes
+ */
+class StaticLinkageChecker {
+
+  /**
+   * Given list of the jar files as file names in filesystem, outputs the report of static linkage
+   * check.
+   *
+   * @param arguments jar files available in the file system
+   * @throws IOException when there is a problem in reading a jar file
+   * @throws ClassNotFoundException when there is a problem in reading a class from a jar file
+   */
+  public static void main(String[] arguments) throws IOException, ClassNotFoundException {
+    StringBuilder stringBuilder = new StringBuilder();
+    List<Path> jarFilePaths = Arrays.asList(arguments).stream()
+        .map(name -> (Paths.get(name)).toAbsolutePath())
+        .collect(Collectors.toList());
+    List<FullyQualifiedMethodSignature> unresolvedMethodReferences =
+        findUnresolvedMethodReferencesInJars(jarFilePaths);
```

Comment:
I think findUnresolvedMethodReferences is enough. No need to duplicate what the arguments already say in the method name.

Reply:
Done